source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -3658,16 +3709,30 @@ void Graph::propertyEditor()
             ImGui::Checkbox("Show output connections", &_currUiNode->_showOutputsInEditor);
 
             int count = 0;
+            float totalImagePadding = 0.0f;
+            float imagePadding = 0.0f;
+            if (_previewSize > 0.0f)
+            {
+                imagePadding = (_previewSize > availableWidth) ? availableWidth : _previewSize;
+            }
             for (UiPinPtr input : _currUiNode->inputPins)
             {
                 if (_currUiNode->_showAllInputs || (input->getConnected() || _currUiNode->getNode()->getInput(input->_name)))
                 {
                     count++;
+                    
+                    // Add space for image previews
```

Comment:
Since we are using a fixed height, we need to take into consideration space for image previews.